design/observers.md

```diff
@@ -0,0 +1,211 @@
+As of now we have Validator Nodes only responsible for write and read requests as well as catching up other Nodes and clients.
+With a state proof support, we can create additional rings of Observer Nodes that can process read requests and reduce load on Validator nodes.
+
+## Goals (in priority order)
+
+- Goal1: Reduce Read Load
+    - Reduce read load on Validators
+
+- Goal2: More Nodes
+    - Have a big number of Nodes available for read requests (we have an essential limit on the size of Validator Nodes because of how chatty RBFT is)
+    
+- Goal3: Reduce Catch-up Load
+    - Reduce catch-up load on Validators
+    
+- Goal4: Reduce Write Load
+    - Be able to send write requests to one Node only (either Validator or Observer)
+    
+- Goal5: Promotion
+    - Be able to replace suspicious Validators to Gatekeepers (promote Gatekeepers)
+    
+- Goal6: Untrusted Observers
+    - Be able to have unlimited number of Nodes with a copy of the ledger
+    
+- Goal7: Fast Catch-up of clients
+    - Be able to bootstrap catch-up for clients using one Node (either Validator or Observer)
+    
+- Goal8: Fast Catch-up of Nodes 
+    - Be able to bootstrap catch-up for Nodes using one Node (either Validator or Observer)
+    
+- Goal9: Close the gates
+    - Forbid all external client communication for Validators (allow only restricted communication with Gatekeepers and other Validators), so that
+        - Gatekeepers ring serves as one of the stoppers for DoS/Spam attacks.
+        - Helps to avoid attacks on Primary
+        - Helps to reduce the number of connections for Validators
+        - Helps to improve availability of Validators
+
+## Architecture
+![Observer Architecture](Observer Architecture.jpg)
+
+###Assumptions
+- We call Trusted Observers “Gatekeepers” because they eventually will become them when we “close the gate” to Validators.
+If we don’t want that to ever happen, then we probably should find a better name for Gatekeepers to avoid confusion (just call them Observers?)
+
+- Gatekeepers are added in permissioned fashion (there is a Steward for each Gatekeeper)
+
+- Gatekeepers are in sync with Validators
+    - It means that clients (indy-sdk) do not need to have neither a separate pool ledger for each Gatekeeper nor a Pool State trie (Patricia trie).
+- Observers may not be fully in sync with Validators and Gatekeepers at a given moment of time (depends on the synchronization policy), although shoud be in sync eventually.
+ There will probably be a gap between the latest Validators state and latest Observer state.
+    - It means that clients (indy-sdk) need to have either a separate pool ledger for each Observer, or a Pool State trie (Patricia trie)
```

Comment:
I mean that Observers can be out of date a bit comparing to other Validators and Observers. In other words, observers may be at a different state at a given time => Observers may have different pool states.
So, a client needs to be able to work with different Observers in their current state => clients may face different pool state.
If clients send query requests to different observers, they may face different state of the pools (needed to verify BLS multisig), so we need to have Patricia merkle state Trie for clients.

Reply:
`different Observers in their current state => clients may face different pool state.` But pool state just  describes the number and identity of nodes (validators/observers)

Reply:
The client may talk with Observers only, and not with Validators. Let's assume the client sends request X to Observer1 which has pool state X1 (this state can be used to validate state proofs). Then it sends request to Observer2 which has an *earlier* state X2 with different values of BLS keys (this state can be used to validate state proofs). So, the client needs to be able to get an old state of the pool to validate BLS sigs. It can easily be done using State trie, that's why I say that we need Pool State Trie for clients (sdk side).